api/middleware/errors.go

```diff
@@ -4,3 +4,9 @@ import "errors"
 
 // ErrMalformedToken signals that a malformed token has been provided
 var ErrMalformedToken = errors.New("malformed token")
+
+// ErrUnknownContentLength signals that the content length is unknown
+var ErrUnknownContentLength = errors.New("unknown content length")
```

Comment:
and reuse it both ErrUnknownContentLength and ErrContentLengthTooLarge

Reply:
I added the uknown use case, because of this.

`// ContentLength records the length of the associated content.
	// The value -1 indicates that the length is unknown.
	// Values >= 0 indicate that the given number of bytes may
	// be read from Body.
	//
	// For client requests, a value of 0 with a non-nil Body is
	// also treated as unknown.
	ContentLength [int64](https://pkg.go.dev/builtin#int64)`
	
	https://pkg.go.dev/net/http#Request.ContentLength